File: src/main/java/org/parham/seasonjob/commands/default_cmd/Accept.java
package org.parham.seasonjob.commands.default_cmd;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.parham.seasonjob.commands.SeasonCommand;
import org.parham.seasonjob.data.member.MemberManager;
import org.parham.seasonjob.data.sender.Messages;

import java.util.*;

public class Accept implements SeasonCommand {
    @Override
    public boolean hasPermission() {
        return false;
    }

    @Override
    public boolean isPlayerOnly() {
        return true;
    }

    @Override
    public String getPermission() {
        return "seasonjob.accept";
    }

    @Override
    public int getNeedArguments() {
        return 0;
    }

    @Override
    public String getUsage() {
        return "";
    }

    @Override
    public void execute(CommandSender sender, Command cmd, String label, String[] args, boolean isLeader, boolean hasAccess) {
        Player player = (Player) sender;
        if (!Invite.invites.containsKey(player.getUniqueId())) {
            player.sendMessage(Messages.getMessage("doesnt-have-invite-accept"));
        } else if (MemberManager.contains(player.getUniqueId())) {
            player.sendMessage(Messages.getMessage("employed-accept"));
        } else {
            Map<UUID, ArrayList<String>> invites = Invite.invites;

            String job;

            if (args.length > 0 && invites.get(player.getUniqueId()).contains(args[0])) {
                job = args[0];
            } else {
                job = invites.get(player.getUniqueId()).get(invites.get(player.getUniqueId()).size() - 1);
            }

            MemberManager.add(player.getUniqueId(), job);
            player.sendMessage(Messages.getMessage("accepted-accept"));

            Invite.invites.remove(player.getUniqueId());
        }
    }

    @Override
    public List<String> getCompletions(CommandSender sender, String[] args) {
        if (sender instanceof Player) {
            return Invite.invites.get(((Player) sender).getUniqueId());
        } else {
            return Collections.emptyList();
        }
    }
}

File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/Accept.kt
package ir.parham.seasonJob.Commands.Default

import Libs.API.ir.parham.SeasonJobsAPI.Commands
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message
import ir.parham.SeasonJobsAPI.Actions.Member
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*
import kotlin.collections.List

class Accept : Commands {
    override fun runner(sender: CommandSender, args: Array<out String>) {
        val message = Message()

        if (sender !is Player) {
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "denyConsole"))
            // deny console
        } else if (!sender.hasPermission("seasonjobs.accept")) {
            sender.sendMessage(message.get(sender, "denyPerm"))
            // deny perm
        } else if (!Invite.invite.containsKey(sender.uniqueId)) {
            sender.sendMessage(message.get(sender, "emptyInvite"))
            // no invite
        } else {
            adder(sender)
            remover(sender)
            sender.sendMessage(message.get(sender, "jobAccepted"))
            // accepted
        }
    }

    override fun completer(sender: CommandSender, args: Array<out String>): List<String> {
        return listOf("")
    }

    override fun remover(target: OfflinePlayer) {
        Invite.invite.remove(target.uniqueId)
    }

    override fun adder(target: OfflinePlayer) {
        Member().remove(target.uniqueId)
        Member(Invite.inviters.get(target.uniqueId).toString()).add(target.uniqueId, Invite.invite.get(target.uniqueId)!!)
    }

    override fun setter(target: OfflinePlayer) {
        TODO("Not yet implemented")
    }
}